lib/util/url.spec.ts

```diff
@@ -79,6 +81,13 @@ describe('util/url', () => {
     expect(trimTrailingSlash('foo//////')).toBe('foo');
   });
 
+  it('ensureTrailingSlash', () => {
+    expect(ensureTrailingSlash('foo')).toBe('foo/');
+    expect(ensureTrailingSlash('/foo/bar')).toBe('/foo/bar/');
+    expect(ensureTrailingSlash('foo/')).toBe('foo/');
+    expect(ensureTrailingSlash('foo//////')).toBe('foo//////');
```

Comment:
@rarkins I think it should behave like:
```suggestion
    expect(ensureTrailingSlash('foo//////')).toBe('foo/');
```

Reply:
The current `ensureTrailingSlash` is `/\/?$/` the `?` only matches lazily(ie; 0 or 1 match and) so it won't match the rest of the slashes in above case.

Reply:
@viceice depends if you think this function is "ensure a trailing slash exists" or "ensure only one trailing slash exists"